grakn-engine/src/main/java/ai/grakn/engine/controller/GraqlController.java

```diff
@@ -102,16 +102,19 @@ private Json executeGraql(Request request, Response response){
         String queryString = mandatoryQueryParameter(request, QUERY);
         boolean infer = parseBoolean(mandatoryQueryParameter(request, INFER));
         boolean materialise = parseBoolean(mandatoryQueryParameter(request, MATERIALISE));
+        Json responseBody;
 
-        try(GraknGraph graph = factory.getGraph(keyspace, READ)){
+        try(GraknGraph graph = factory.getGraph(keyspace, WRITE)){
             Query<?> query = graph.graql().materialise(materialise).infer(infer).parse(queryString);
 
             if(!readOnly(query)){
                 throw new GraknEngineServerException(405, "Only \"read-only\" queries are allowed.");
             }
-
-            return respond(request, query, response);
+            //Do not return directly inside the try block otherwise open transactions will not autocommit and close properly.
+            responseBody = respond(request, query, response);
```

Comment:
With READ tx, the reasoner inference throws an exception when trying to materialise because the graph is read-only. We changed it to WRITE and it stopped throwing exceptions, but then the IDs returned to the client, computed by Reasoner, were not materialised in the graph, so we figured the tx was not committing properly. Moving the return outside the try-with-resource block solved it